Empty services

y-lohse opened the pull request:
This PR replaces the component we show when the user has no configured connectors.

<img width="784" alt="Capture d'écran 2019-11-12 16 55 26" src="https://user-images.githubusercontent.com/2261445/68689733-221d2480-0571-11ea-84a2-f3dc4531a8ef.png">

Still requires 
- https://github.com/cozy/cozy-ui/pull/1240 ☑
- a new cozy-ui version ☑
- pushing translations

But ready for review!

ptbrowne commented on src/styles/lists.styl:
```diff
@@ -175,6 +175,23 @@ heavyShadow = 0 4px 24px 0 rgba(50, 54, 63, .08)
+
+    +tiny-screen()
+        padding-left .5rem
+
```
It seems that in this repo, we do not use the suit CSS convention that we follow (loosely) across our repos (https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md). Is there a good reason not to folllow it ? do you think we should use it for new code ?

y-lohse replied:
I'l have a proper look at it, but it's probably just me going with the existing convention I see in the code. And yeah def. makes sense to follow our normal conventions for new code 👍

ptbrowne commented:
Did not check the code but it seems that the margin-bottom of the "Start gathering your data" is a bit big, it seems the text is more connected to the above icons than to the explanation text.

Crash-- commented on src/components/CandidateCategoryTile.jsx:
```diff
@@ -0,0 +1,59 @@
+import AppLinker, { generateWebLink } from 'cozy-ui/react/AppLinker'
+import { withClient } from 'cozy-client'
+
+class CandidateCategoryTile extends React.Component {
```
Since this component is just a render do we have some advantages to keep in as a Class and not a function? Since we're doing more and more hooks, I'm wondering if we should start a new habit

y-lohse replied:
The reason I used a class was to be abble to use context variables for domain and secure, but I'll change that and then we can use simple functions 👍

Crash-- commented on src/components/CandidateCategoryTile.jsx:
```diff
@@ -0,0 +1,59 @@
+                    key={slug}
+                    className="item-grid-icon"
+                    domain={domain}
+                    secure={secure}
```
AppIcon should be able to only take a `cozyURL` props (coming from new URL(client.getStackClient().uri) for instance) and not deal with `secure` and `domain` no ?

y-lohse replied:
I think so too, I'm not sure why it takes 2 distinct props.

Crash-- commented on src/components/CandidateCategoryTile.jsx:
```diff
@@ -0,0 +1,59 @@
+        href={generateWebLink({
+          cozyUrl: cozyURL.origin,
+          slug: app,
+          nativePath: nativePath
```
We should also pass `nested` true/false?

ptbrowne commented on src/components/CandidateCategoryTile.jsx:
```diff
@@ -0,0 +1,59 @@
+  client: PropTypes.object.isRequired,
+  domain: PropTypes.string.isRequired,
+  secure: PropTypes.bool
+}
```
I do not think we should use the legacy context API since we are not sure it will work in upcoming versions of React. Are you sure there is no other way around using the context ? It seems here that there is a lot of props, maybe you wanted to use propTypes ?

ptbrowne commented on src/components/CandidateServiceTile.jsx:
```diff
@@ -0,0 +1,52 @@
+class CandidateServiceTile extends React.Component {
+  render() {
+    const { t, slug, name, client } = this.props
+    const { domain, secure } = this.context
```
domain and secure can be deduced from the client. This way we do not have to touch the context manually.

y-lohse commented:
I added a wrapper component for AppIcon that injects the `secure` and `domain` props so we don't have to rely on the context API anymore. I also turned some class components into simple functions, and used our regular css naming convention on the classes I added!

ptbrowne commented on src/components/AppIcon.jsx:
```diff
@@ -0,0 +1,12 @@
+  return <CozyUiAppIcon domain={domain} secure={secure} {...otherProps} />
+}
+
+export default withClient(AppIcon)
```
Could you add a TODO to move this to cozy-ui ?
